htlcswitch/hop/error_encryptor.go

```diff
@@ -91,11 +106,29 @@ type SphinxErrorEncrypter struct {
 //  1. Decode: to deserialize the ephemeral public key.
 //  2. Reextract: to "unlock" the actual error encrypter using an active
 //     OnionProcessor.
-func NewSphinxErrorEncrypter() *SphinxErrorEncrypter {
-	return &SphinxErrorEncrypter{
+func NewSphinxErrorEncrypter(encType EncrypterType) *SphinxErrorEncrypter {
+	encrypter := &SphinxErrorEncrypter{
 		OnionErrorEncrypter: nil,
 		EphemeralKey:        &btcec.PublicKey{},
 	}
+
+	// Fill in additional fields for encrypters in a blinded route.
+	switch encType {
+	case EncrypterTypeIntroduction:
+		encrypter.BlindingKey = BlindingKey{
+			blindingPoint:  &btcec.PublicKey{},
```

Comment:
Why set up this wrapped public key, only to leave it empty? Will we read the key data later? Yes, this is handled in the error encryptor's `decode()` method.